src/Components/Tasks/Tasks.tsx

```diff
@@ -1,129 +1,216 @@
+  const [modalState, setModalState] = useState("close")
+  const [modalUpdate, setModalUpdate] = useState("close")
   const handleClose = () => setModalState("close");
-  const { baseUrl,reqHeaders,role }: any = useContext(AuthContext);
-  // console.log(role);
-  
+  const [projectId, setProjectId] = useState(0);
+  const { baseUrl, reqHeaders, role }: any = useContext(AuthContext);
+  const {getToastValue}=useContext(ToastContext)
   const navigate = useNavigate();
+ 
+  const [todos,setTodos]=useState([])
+  const [inProgress,setInProgress]=useState([])
+  const [done,setDone]=useState([])
+  
+ 
+  
+
+  {/* show All Tasks for Manager  */ }
+   const getTasksList = ( ) => {
+    axios
+      .get(`${baseUrl}/Task/manager?pageSize=14&pageNumber=1`, { headers: reqHeaders })
+      .then((response: any) => {
+        console.log(response.data.data);
+        setTasks(response?.data?.data);
+
+      })
+      .catch((error: any) => {
+        getToastValue("error", error.response.data.data.message);
 
-  const navigateTo = () => {
-    navigate("/dashboard/add-task");
+      });
   };
 
-  const addToTask =()=>{
+ {/*Navigate to Add New Task by Manager  */ }
+  const navToAddTask = () => {
     navigate('/dashboard/add-task')
   }
-  const showDeleteModel = ()=>{
-   
-    
+
+  {/* show Delete Task Model */ }
+  const showDeleteModel = (id) => {
     setModalState("delete-modal")
+    setProjectId(id)
   }
 
-  const showUpdateModel = ()=>{
-   
-    
-    setModalUpdate("update-modal")
+  {/*Delete Task */ }
+  const deleteTask = () => {
+    axios.delete(`${baseUrl}/Task/${projectId}`, { headers: reqHeaders }).then((response) => {
+      getToastValue('success','Deleted Successfully!!')
+      handleClose();
+      getTasksList()
+    }).catch((error) => {
+      getToastValue("error",error?.response?.data?.message)
+    })
   }
 
-  const getTasksList = () => {
-    axios
-      .get(`${baseUrl}/Task/manager`,{ headers: reqHeaders })
-      .then((response: any) => {
-        console.log(response.data.data);
-        setTasks(response?.data?.data);
+  {/*Update Task */ }
+  const editTaskShow =(id)=>{
+    
+    navigate(`/dashboard/edit-task/${id}`)
 
-      })
-      .catch((error:any) => {
-        console.log("error",error.response.data.data.message);
+  }
 
-      });
-  };
+  const getEmployeeTasks =()=>{
+    axios.get(`${baseUrl}/Task`,{headers:reqHeaders}).then((response)=>{
+      console.log(response);
+      
+  
+      const todosTask=response?.data?.data.filter((task:any)=>task.status=='ToDo')
+      const inProgressTask=response?.data?.data.filter((task:any)=>task.status=='InProgress')
+      const doneTask=response?.data?.data.filter((task:any)=>task.status=='Done')
+      setTodos(todosTask)
+      setInProgress(inProgressTask)
+      setDone(doneTask)
+  
+      setTasks(response.data.data)
+   
+    })
+   }
+  
 
   useEffect(() => {
+    if(role==='Manager')
     getTasksList();
+  else
+  getEmployeeTasks()
```

Comment:
If we login as a manager and  go the tasks page and  refreshit, the initial value of the role is null from the context, which will make  this condition run, and since we don't have the `role` in the dependency array, we will never run `getTasksList` again unless we go to another page and then return back to the tasks page.

To fix this, we need to make the second condition explicit:
 `else if(role==="Manager")`

and add the `role` in the dependency array.
`  }, [role]);`

This will make the effect run whenever the role changes.

![Error on reloading](https://github.com/FatmaHossam5/project-managment-system-butterCup-/assets/64327685/ac927e95-62e5-447f-9bf7-5c825e09594c)